modules/n1ql/pages/n1ql-language-reference/altersequence.adoc

```diff
@@ -0,0 +1,416 @@
+
+The namespace, bucket, scope, and sequence name must follow the rules for {identifiers}[identifiers].
+If the namespace, bucket, scope, or sequence name contain any special characters such as hyphens (-), you must wrap that part of the expression in backticks ({backtick} {backtick}).
+
+[[alter-sequence-options]]
+=== Sequence Options
+
+[source,ebnf]
+----
+include::partial$grammar/ddl.ebnf[tag=alter-sequence-options]
+----
+
+image::n1ql-language-reference/alter-sequence-options.png["Syntax diagram: refer to source code listing", align=left]
+
+You can use the following optional clauses to specify individual attributes for the sequence.
+
+[horizontal.compact]
+restart-with:: <<restart-with>> icon:caret-down[]
+increment-by:: <<increment-by>> icon:caret-down[]
+maxvalue:: <<maxvalue>> icon:caret-down[]
+minvalue:: <<minvalue>> icon:caret-down[]
+cycle:: <<cycle>> icon:caret-down[]
+cache:: <<cache>> icon:caret-down[]
+
+[[restart-with]]
+==== RESTART WITH Clause
+
+[source,ebnf]
+----
+include::partial$grammar/ddl.ebnf[tag=restart-with]
+----
+
+image::n1ql-language-reference/restart-with.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the RESTART keyword by itself to restart the sequence from its original starting value.
+
+Use the RESTART WITH clause to restart the sequence from a new value.
+
+If this clause is omitted, the sequence does not restart.
+
+[horizontal#restart-with-args]
+integer::
+(Optional) The restart value for the sequence.
+
+[[increment-by]]
+==== INCREMENT BY Clause
+
+[source,ebnf]
+----
+include::partial$grammar/ddl.ebnf[tag=increment-by]
+----
+
+image::n1ql-language-reference/increment-by.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the INCREMENT BY clause to specify the incremental value of each step in the sequence.
+
+If this clause is omitted, the incremental value is not altered.
+
+[horizontal#increment-by-args]
+integer::
+(Required) The step size for the sequence.
+Use a negative value for a descending sequence.
+
+[[maxvalue]]
+==== MAXVALUE Clause
+
+[source,ebnf]
+----
+include::partial$grammar/ddl.ebnf[tag=maxvalue]
+----
+
+image::n1ql-language-reference/maxvalue.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the MAXVALUE clause to specify the maximum value for the sequence.
+
+Use NO MAXVALUE to specify that the maximum value is the highest signed 64-bit integer, `2^63^-1`.
+
+If this clause is omitted, the maximum value is not altered.
+
+[horizontal#maxvalue-args]
+integer::
+(Optional) The maximum value for the sequence.
+
+[[minvalue]]
+==== MINVALUE Clause
+
+[source,ebnf]
+----
+include::partial$grammar/ddl.ebnf[tag=minvalue]
+----
+
+image::n1ql-language-reference/minvalue.png["Syntax diagram: refer to source code listing", align=left]
+
+Use the MINVALUE clause to specify the minimum value for the sequence.
+
+Use NO MINVALUE to specify that the minimum value is the lowest signed 64-bit integer, `-2^63^-1`.
```

Comment:
-2 to the 63; no -1 for the minimum (-9223372036854775808).

Reply:
Done